src/stan/lang/ast/scope.hpp

```diff
@@ -66,6 +66,15 @@ namespace stan {
       bool is_local() const;
 
       /**
+       * Flags local scopes which permit parameter variables,
```

Comment:
I'd suggest following the definition in the doc and saying it's a local variable that is not defined in the transformed data or generated quantities block.

That would allow it to be defined in an arbitrary function, right, or should a `data` specification rule that out?